backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/codec/impl/LuceneFieldCodec.java

```diff
@@ -54,4 +55,8 @@ public interface LuceneFieldCodec<F> {
 	 * in doubt.
 	 */
 	boolean isCompatibleWith(LuceneFieldCodec<?> other);
+
+	default KnnVectorsFormat knnVectorFormat() {
+		return null;
+	}
```

Comment:
Does this need to be this high in the hierarchy? It really doesn't seem relevant to anything but vector fields.

Reply:
I've moved it to a "vector-specific" codec. My initial idea to put it there at the very top was to prevent from various `intsanceof` checks 😃. Ideally I'd put it inside of the field codec itself and pass the filter codec in, but I didn't see how to do that to make it work, hence this "getter" 😃

Reply:
Wouldn't it make more sense to retrieve the vector format directly from the field type, like we do for e.g. `org.hibernate.search.backend.lucene.types.impl.LuceneIndexValueFieldType#indexingAnalyzerOrNormalizer`?